storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -6998,6 +7012,206 @@ xb_init()
 	return(true);
 }
 
+static const char* replace_default_name(const char* name)
+{
+	if (strcmp(name, "*") == 0) {
+		return "\\*";
+	}
+	return name;
+}
+
+static bool
+has_privilege(const std::list<std::string> &granted,
+	const char* required,
+	const char* db_name,
+	const char* table_name)
+{
+	char buffer[1000];
+	xb_regex_t priv_re;
+	xb_regmatch_t tables_regmatch[1];
+	bool result = false;
+
+	db_name = replace_default_name(db_name);
+	table_name = replace_default_name(table_name);
+
+	int written = snprintf(buffer, sizeof(buffer),
+		"GRANT .*(%s)|(ALL PRIVILEGES).* ON (\\*|`%s`)\\.(\\*|`%s`)",
+		required, db_name, table_name);
+	if (written < 0 || written == sizeof(buffer)
+		|| !compile_regex(buffer, "check_privilege", &priv_re)) {
+		exit(EXIT_FAILURE);
+	}
+
+	typedef std::list<std::string>::const_iterator string_iter;
+	for (string_iter i = granted.begin(), e = granted.end(); i != e; ++i) {
+		int res = xb_regexec(&priv_re, i->c_str(),
+			1, tables_regmatch, 0);
+
+		if (res != REG_NOMATCH) {
+			result = true;
+			break;
+		}
+	}
+
+	xb_regfree(&priv_re);
+	return result;
+}
+
+enum {
+	PRIVILEGE_OK = 0,
+	PRIVILEGE_WARNING = 1,
+	PRIVILEGE_ERROR = 2,
+};
+
+static
+int check_privilege(const std::list<std::string> &granted_priv,
```

Comment:
Yeah, comments are different now, but code formatting remains the same, argument list starts on new line. I don't mind to put all arguments in one line, but if each occupies separate line, they should follow InnoDB conventions.